freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/UserSyncService.java

```diff
@@ -358,14 +351,70 @@ private SyncStatusDetail internalSynchronizeStack(Stack stack, UmsUsersState ums
                 LOGGER.debug("Finished {}.", LogEvent.SYNC_CLOUD_IDENTITIES);
             }
 
+            retrySyncIfBatchCallHasWarnings(stack, umsUsersState, fullSync, warnings, freeIpaClient,
```

Comment:
this should probably move above the cloud identity sync since the retry does not affect cloud identities.